symbolic-il2cpp/src/line_mapping/mod.rs

```diff
@@ -116,20 +119,170 @@ mod tests {
             // 13
             //<source_info:main.cs:29>
             actual source code // 15
-        ";
-
-        let line_mappings: HashMap<_, _> = SourceInfos::new(cpp_source)
-            .map(|si| (si.cpp_line, si.cs_line))
-            .collect();
-
-        let mapping = HashMap::from([("main.cpp", HashMap::from([("main.cs", line_mappings)]))]);
-        let mapping_json = serde_json::to_string(&mapping).unwrap();
-
-        let parsed_mapping = LineMapping::parse(mapping_json.as_bytes()).unwrap();
-
+        ",
+        );
+        assert_eq!(parsed_mapping.lookup("main.cpp", 0), None);
+        assert_eq!(parsed_mapping.lookup("main.cpp", 1), None);
         assert_eq!(parsed_mapping.lookup("main.cpp", 2), None);
+        assert_eq!(parsed_mapping.lookup("main.cpp", 3), None);
+        assert_eq!(parsed_mapping.lookup("main.cpp", 4), None);
         assert_eq!(parsed_mapping.lookup("main.cpp", 5), Some(("main.cs", 17)));
-        assert_eq!(parsed_mapping.lookup("main.cpp", 12), None);
+        assert_eq!(parsed_mapping.lookup("main.cpp", 6), Some(("main.cs", 17)));
```

Comment:
this might be a bit overkill here to enumerate every line. the important one I guess is that `12` now has a mapping.

Reply:
it may be but it also makes the expectation clear (and removes the guesswork) so if you don't mind I'd keep it there for each line